libcontainer/cgroups/fscommon/utils_test.go

```diff
@@ -101,3 +107,53 @@ func TestGetCgroupParamsInt(t *testing.T) {
 		t.Fatal("Expecting error, got none")
 	}
 }
+
+type cgroupTestUtil struct {
+	// cgroup data to use in tests.
+	CgroupData *cgroupData
+
+	// Path to the mock cgroup directory.
+	CgroupPath string
+
+	// Temporary directory to store mock cgroup filesystem.
+	tempDir string
+	t       *testing.T
+}
+
+// Creates a new test util for the specified subsystem
+func NewCgroupTestUtil(subsystem string, t *testing.T) *cgroupTestUtil {
+	d := &cgroupData{
+		config: &configs.Cgroup{},
+	}
+	d.config.Resources = &configs.Resources{}
+	tempDir, err := ioutil.TempDir("", "cgroup_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	d.root = tempDir
+	testCgroupPath := filepath.Join(d.root, subsystem)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure the full mock cgroup path exists.
+	err = os.MkdirAll(testCgroupPath, 0o755)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &cgroupTestUtil{CgroupData: d, CgroupPath: testCgroupPath, tempDir: tempDir, t: t}
+}
+
+func (c *cgroupTestUtil) cleanup() {
+	os.RemoveAll(c.tempDir)
+}
+
+// Write the specified contents on the mock of the specified cgroup files.
+func (c *cgroupTestUtil) writeFileContents(fileContents map[string]string) {
+	for file, contents := range fileContents {
+		err := cgroups.WriteFile(c.CgroupPath, file, contents)
+		if err != nil {
+			c.t.Fatal(err)
+		}
+	}
+}
```

Comment:
OK I haven't realized you were copying all this from `fs` package. This is not good. For one thing, the implementation in `fs` has already improved. That does not mean that you need to copy those improvements -- this is not going to work in the long term.

I think since both fs and fscommon tests need these helpers, they need to be moved to cgroups package (somewhat similar to what I did in #3024).

Reply:
@kolyshkin Sure i can move this to `cgroup` I'll make relevant changes and re-base

Reply:
@kolyshkin Should i create a separate PR for refactor of TestUtils or seperate commit in same PR cause it contains changes which are out-of-scope wrt to this pull request.

Reply:
@kolyshkin i think separate PR would be better for discussions https://github.com/opencontainers/runc/pull/3122 since too many changes are there.

Reply:
OK, this was a very bad idea of mine.

I took a look at the test you are adding -- in practice, it seems that having something like

```go
root := t.TempDir()
rdmaDir := filepath.Join(root, "rdma")
 if err := os.MkdirAll(testCgroupPath, 0o755); err != nil {
                t.Fatal(err)
        }
```

should be enough for the test case you're adding.

Reply:
@kolyshkin Should i drop changes  in `libcontainer/cgroups/fscommon/utils_test.go` and have this minimal block in `libcontainer/cgroups/fscommon/rdma_test.go` itself only, WDYT ? however this might be inconsistent with how tests for other subsystems are written.

Reply:
@kolyshkin Done made changes and re-based with upstream please take a look again.

Reply:
@kolyshkin dropped changes for `libcontainer/cgroups/fscommon/utils_test.go` everything stays with RDMA tests itself. PTAL.